src/api.ts

```diff
@@ -92,10 +94,12 @@ export interface FernsRouterOptions<T> {
    * These can be overridden by the user if not disallowed by queryFilter. */
   defaultQueryParams?: {[key: string]: any};
   /** Paths to populate before returning data from list queries. Accepts Mongoose-style populate strings.
+   *  May also be a function that takes the request and returns a list of paths to populate. This is handy if you need
+   *  to populate based on the user or request, such as app version.
    *    ["ownerId"] // populates the User that matches `ownerId`
    *    ["ownerId.organizationId"] // Nested. Populates the User that matches `ownerId`, as well as their organization.
    * */
-  populatePaths?: string[];
+  populatePaths?: string[] | ((req: express.Request) => string[]);
```

Comment:
At first I wasn't sold that this should be required to be an Express request type instead of a more generic function, but I think because of how we are using it directly with `fernsRouter` as basically a type of middleware it makes sense.

Reply:
Agreed, it's pretty dang close to Express middleware. I was half tempted to pass in `res` too, but couldn't think of a reason you'd want that.